torch/distributed/rpc.py

```diff
@@ -0,0 +1,104 @@
+from . import invoke_rpc
+from . import ProcessGroupAgent
+
+import array
+import torch
+
+_agent = None
+
+def _collect_worker_names(name, group):
+    from . import all_gather
+    from . import get_world_size
+
+    # collect name length
+    ws = get_world_size(group)
+    name_len = len(name)
+    len_input = torch.ones(1, dtype=torch.int64) * name_len
+    len_outputs = [torch.empty(1, dtype=torch.int64) for _ in range(ws)]
+    all_gather(len_outputs, len_input, group=group)
+
+    # collect name value
+    max_len = torch.stack(len_outputs).max().item()
+    name_input = torch.empty(max_len, dtype=torch.uint8)
+    name_input[:name_len] = torch.Tensor(list(name.encode('utf-8')))
+    name_outputs = [torch.empty(max_len, dtype=torch.uint8) for _ in range(ws)]
+    all_gather(name_outputs, name_input, group=group)
+
+    names = []
+    for i in range(ws):
+        name_tensor = name_outputs[i][:len_outputs[i]]
+        names.append(array.array('B', name_tensor.tolist()).tobytes().decode('utf-8'))
+
+    return names
+
+
+def init_rpc(name, backend='pg'):
+    r"""
+    Arguments:
+        name (str): name of this worker.
```

Comment:
Any advice for how to actually fill out this field? I assume each worker needs a unique name? (Why isn't this already handled by `init_process_gorup`?)

Reply:
Will add docs. Yes, every worker needs a unique name. I think there are two reasons for not letting `init_process_group` taking a name. 

1. For most c10d/ddp features, all participants are kind of conceptually identical. Having a rank would be sufficient. It does not make much sense to allow names like `ParameterServer` or `Trainer2`.
2. `ProgressGroup` is one of the backend option for `init_rpc`, and there might be other options in the future (e.g., Thrift). So, we would like to have some independent naming in the RPC layer to help users easily identify RPC participants.